app/Platform/Jobs/UpdatePlayerGameMetricsJob.php

```diff
@@ -78,6 +78,30 @@ public function handle(): void
 
         $isBatched = $this->batchId !== null;
 
+        // We might be able to skip a significant amount of processing if this is part of a
+        // batch job and the player has never unlocked any achievements. To confirm, we need to
+        // check for actual unlocks, not just the denormalized counts, because unofficial
+        // achievements don't count towards achievements_unlocked until they're promoted.
+        if ($isBatched && $playerGame->achievements_unlocked === 0 && $playerGame->all_achievements_unlocked === 0) {
+            // Double-check if the player has ANY unlocks for this game (including unofficial).
+            // If they do have any unlocks, we'll run a full metrics update.
+            $hasAnyUnlocks = $playerGame->user->playerAchievements()
+                ->whereHas('achievement', function ($query) {
+                    $query->where('GameID', $this->gameId);
+                })
+                ->exists();
+
+            // If they don't have any unlocks, we'll run a lightweight job just to calculate
+            // the player's playtime, and we'll skip processing all the heavy stuff.
+            if (!$hasAnyUnlocks) {
+                // Always update playtime for players with no achievements.
+                dispatch(new UpdatePlayerGamePlaytimeJob($this->userId, $this->gameId))
+                    ->onQueue('player-game-metrics-batch');
```

Comment:
Why dispatch this? The much more expensive full call is executed without being dispatched on line 106 below. We're already in a dispatched job for the playerGame record at this point.

You could just inline the handle functionality and avoid another job and another query for the playerGame record:
```
if (!$hasAnyUnlocks) {
    $activityService = new PlayerGameActivityService();
    $activityService->initialize($playerGame->user, $playerGame->game);
    $summary = $activityService->summarize();

    // Only save if playtime actually changed.
    if ($playerGame->playtime_total !== $summary['totalPlaytime']) {
        $playerGame->playtime_total = $summary['totalPlaytime'];
        $playerGame->save();
    }
    return;
}
```

Reply:
Yep, that's a lot better. Updated in latest.